core/src/abi.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{ConsensusError, ConsensusState, ensure};
+
+// Defines the Solidity struct layouts for public types
+alloy_sol_types::sol! {
+    struct Checkpoint {
+        uint64 epoch;
+        bytes32 root;
+    }
+
+    struct State {
+        Checkpoint current_justified;
+        Checkpoint finalized;
+    }
+
+    struct Journal {
```

Comment:
Now that we have one single struct for the output, is there still a reason to keep two structs for the input? I think we should unify them.

Reply:
100% agreed on that. See subsequent changes